### ISSUE TITLE
[Bug]: Wrong representation of window

### ISSUE THREAD

**Issue by adrian-widmer** [ThatOpen/engine_web-ifc#824]
### What happened?

I integrated the viewer of That Open Company in our application (Angular application written in TypeScript) and if I load an IFC model that we created in ArchiCAD, two of the windows are shown in a very strange way (see attached images - issue1, issue2). In ArchiCAD as well as in another viewer (BIMCollabZoom), the model looks as expected (see attached image - normal). So, I'm not sure what exactly is wrong in the rendering of this model but there seems to be a bug. Could someone have a look at this or get in touch with me if further information are needed. I could also share the model if that would help, just let me know so that I can share it privately.

Hint: I load it as well on the demo page (https://thatopen.github.io/engine_web-ifc/demo/) and checked how it looks and the issue is the same as in my application.

![issue1](https://github.com/ThatOpen/engine_web-ifc/assets/163395893/8473911e-6315-4d9d-80e6-05417a1b0b3c)
![issue2](https://github.com/ThatOpen/engine_web-ifc/assets/163395893/6181ff8b-30dd-462d-a359-b9cf77165fb0)
![normal](https://github.com/ThatOpen/engine_web-ifc/assets/163395893/e89e1791-e9b7-48dd-9904-ab6fd1dcc62b)

### Version

0.0.53

### What browsers are you seeing the problem on?

Chrome

**Comment by QuimMoya** [comment]
Can you share the model?

**Comment by adrian-widmer** [comment]
Yes, sure. Is there an easy way I can share the model with you privately?

**Comment by QuimMoya** [comment]
Please send it to support@thatopen.com, specifying the issue it relates to, and mention that this model is for @QuimMoya.

**Comment by adrian-widmer** [comment]
Ok, sure will send it right away. Thanks.

**Comment by QuimMoya** [comment]
Last improvements in fuzzy bools library greatly reduced the issues in your model, still some bugs anyway.
![image](https://github.com/ThatOpen/engine_web-ifc/assets/12625508/18bdceb4-a862-43be-99a0-bcdbe22d812b)

**Comment by adrian-widmer** [comment]
@QuimMoya Thanks for the input and sorry for getting back to you that late but I either didn't see the notification or it went to my spam somehow. So, that means you improved already a bit in this direction. I actually just had similar if not even more severe viewing issues recently where a model looked like this:
![Screenshot 2024-09-12 at 10 35 18](https://github.com/user-attachments/assets/d95ac1dd-2ac2-4b5c-bc0f-a1198e7a98fe)

So, it seems that several things can't be shown correctly. Since you already started to improve several things, do you know why it looks so weird?

...because this is quite a blocker for us because customers will not gain any trust in a viewer that shows models in such a funky way

I will send you the file again via mail so that you have everything you need to localise the issue.

**Comment by adrian-widmer** [comment]
@QuimMoya Hey a guy from the That Open Community had a look into the issue and gave the following feedback that might also help in this case:
I did take a quick look, I'm not sure what the cause is
- I ruled out normals by rendering both sides of all faces (it looks like on https://thatopen.github.io/engine_web-ifc/demo/) (geometry still broken)
- I get no web-ifc errors when loading the file
- It looks fine in blender, although the triangulation is a little weird
- The broken geometry can be traced back to IFCPOLYGONALFACESET so I'm thinking that either:
- the web-ifc triangulation has a bug https://github.com/ThatOpen/engine_web-ifc/blob/8179ec4ca51624989bdf9289b8c8e326d042ca2f/src/cpp/geometry/IfcGeometryProcessor.cpp#L554
- or the exported polygons don't respect the conditions here: https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/schema/ifcgeometricmodelresource/lexical/ifcpolygonalfaceset.htm

I hope that helps ;-)

**Comment by agviegas** [comment]
Model privately available [here](https://drive.google.com/file/d/16MAieQAlrHFW_BTYm9kwHf5igduOlu5I/view?usp=drive_link)

**Comment by adrian-widmer** [comment]
Hey @QuimMoya, did you have the chance to look into this issue by any chance?
It seems that quite some people have similar issues with this. If I can help in any way let me know.

**Comment by HoyosJuan** [comment]
I relate [here](https://people.thatopen.com/c/ask-the-community/wrong-rendering-of-various-objects#comment_wrapper_44894994) the conversation in the community.

**Comment by QuimMoya** [comment]
![image](https://github.com/user-attachments/assets/77373901-8c28-4d24-b09e-7bb6e9c6b06b)
It is improving

**Comment by QuimMoya** [comment]
This model also improved

![image](https://github.com/user-attachments/assets/ce17f068-4181-4eff-bced-c6ac850b7cf2)

**Comment by adrian-widmer** [comment]
@QuimMoya Thanks for the input, that looks much better. Is the updated version already available in a new web-ifc release so that I can test it? (looking forward to it a lot 🤗)

**Comment by adrian-widmer** [comment]
Thanks for the quick reply...will check whether I can update and test it in that case 👍🏻

**Comment by agviegas** [comment]
Not all changes have been published yet. We'll publish them asap!

**Comment by adrian-widmer** [comment]
@agviegas Ok, thanks for the reply. Can you tell already in what version of web-ifc all the changes will be included so that I check for that version before I do my testing?

**Comment by agviegas** [comment]
@adrian-widmer The next one, which is `0.0.60`!

**Comment by adrian-widmer** [comment]
@agviegas Cool, thanks Antonio

**Comment by QuimMoya** [comment]
<img width="977" height="545" alt="Image" src="https://github.com/user-attachments/assets/2a7d93db-7216-455b-a432-49437606e592" />

<img width="966" height="542" alt="Image" src="https://github.com/user-attachments/assets/3411a857-27c7-4823-a88d-704bbe9cb655" />

I think it looks fine now from both sides. I will close the issue
